lib/modules/manager/mix/extract.ts

```diff
@@ -28,27 +36,30 @@ export async function extractPackageFile(
       } while (!contentArr[lineNumber].includes('end'));
       let depMatchGroups = depMatchRegExp.exec(depBuffer)?.groups;
       while (depMatchGroups) {
-        const { depName, datasource, currentValue, organization } =
-          depMatchGroups;
+        const { app, requirement, opts } = depMatchGroups;
+        const github = githubRegexp.exec(opts)?.groups?.value;
+        const git = gitRegexp.exec(opts)?.groups?.value;
+        const ref = refRegexp.exec(opts)?.groups?.value;
+        const branch = branchRegexp.exec(opts)?.groups?.value;
+        const tag = tagRegexp.exec(opts)?.groups?.value;
+        const organization = organizationRegexp.exec(opts)?.groups?.value;
 
-        const dep: PackageDependency = {
-          depName,
-          currentValue,
-        };
+        let dep: PackageDependency;
 
-        dep.datasource = datasource || HexDatasource.id;
-
-        if (dep.datasource === HexDatasource.id) {
-          dep.currentValue = currentValue;
-          dep.packageName = depName;
-        }
-
-        if (organization) {
-          dep.packageName += ':' + organization;
-        }
-
-        if (dep.datasource !== HexDatasource.id) {
-          dep.skipReason = 'non-hex-dep-types';
+        if (git ?? github) {
+          dep = {
+            depName: app,
+            currentValue: ref ?? branch ?? tag,
```

Comment:
- ref needs to be in `currentDigest` instead. `currentValue` can be undefined then.